code/modules/power/tesla/energy_ball.dm

```diff
@@ -320,17 +333,30 @@
 
 	else if(closest_mob)
 		var/shock_damage = (tesla_flags & TESLA_MOB_DAMAGE)? (min(round(power/600), 90) + rand(-5, 5)) : 0
-		closest_mob.electrocute_act(shock_damage, source, 1, tesla_shock = 1, stun = (tesla_flags & TESLA_MOB_STUN))
+		if(zap_gib && iscarbon(closest_mob))
+			var/mob/living/carbon/C = closest_mob
+			C.electrocute_act(shock_damage, source, 1, tesla_shock = 1, stun = (tesla_flags & TESLA_MOB_STUN), gib = TRUE)
+		else
+			closest_mob.electrocute_act(shock_damage, source, 1, tesla_shock = 1, stun = (tesla_flags & TESLA_MOB_STUN))
 		if(issilicon(closest_mob))
 			var/mob/living/silicon/S = closest_mob
 			if((tesla_flags & TESLA_MOB_STUN) && (tesla_flags & TESLA_MOB_DAMAGE))
 				S.emp_act(EMP_LIGHT)
-			tesla_zap(S, 7, power / 1.5, tesla_flags, shocked_targets) // metallic folks bounce it further
+			if(zap_gib)
+				tesla_zap(S, 7, power / 1.5, tesla_flags, shocked_targets, TRUE)
+			else
+				tesla_zap(S, 7, power / 1.5, tesla_flags, shocked_targets) // metallic folks bounce it further
```

Comment:
If we're gibbing in electrocute act, do we still need this code in tesla zap?

Reply:
we do need tesla_zap() only

Reply:
lemme fix this

Reply:
we do need tesla_zap() to launch the next zap to another spess